gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -269,6 +280,7 @@ public void commit()
   @Override
   public void write(D record)
       throws IOException {
+    Preconditions.checkState(!closed, "Writer already closed");
```

Comment:
When did you see this edge case happen? And does this cause the fork to immediately terminate?

Reply:
Given there's so much usage and abstract classes which this extends I wanted to be sure that this condition is preserved